gamemodes/terrortown/entities/weapons/weapon_ttt_defibrillator.lua

```diff
@@ -147,6 +151,12 @@ if SERVER then
 			return
 		end
 
+		if ply:Alive() and not (SpecDM and not ply:IsGhost()) then
```

Comment:
Wouldn't this have to be IsTerror instead of IsAlive? Because afaik spectators are also alive

Reply:
I don't think that's true, I've had the similar patch for my mesmerist's defib on my server with no issues.

Reply:
If I recall correctly, a spectator may be dead or alive:
A dead spectator refers to a player who died during the round and is spectating.
An alive spectator refers to a player who is not participating in the round (ex. they joined during the active round, and got put on the spectator team).

There is also the special case of SpecDM, wherein a spectator is resurrected as a "ghost" and is technically alive. This if statement seems to handle that.

Aside from SpecDM, I don't think it's possible for an alive spectator to drop a corpse.

Reply:
Does IsActive() check if the player is alive or just that they are active in the round (not a spectator)?

Reply:
The latter:
```lua
---
-- Checks whether a @{Player} is a terrorist and in an active round
-- @return boolean
-- @realm shared
function plymeta:IsActive()
	return GetRoundState() == ROUND_ACTIVE and self:IsTerror()
end
```

Reply:
Thanks, but just testing it now, IsActive() _does_ seem to check if they are actually alive

Reply:
That would be because of the internal IsTerror() check. Players who aren't alive tend to be spectators, which makes IsTerror() return false.